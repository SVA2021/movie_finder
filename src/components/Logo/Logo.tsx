import { LocalMovies } from '@mui/icons-material';
import { Typography } from '@mui/material';
import { theme } from '../../theme';

const Logo = () => {
	return (
		<Typography data-testid='logo' variant='h1' component={'p'} sx={{
			display: 'flex',
			alignItems: 'center',
			color: theme.palette.text.secondary,
			textTransform: 'uppercase',
			fontFamily: "Roboto,Arial,sans-serif",
		}}>
			Movie
			<LocalMovies sx={{
				width: 'auto',
				height: theme.typography.pxToRem(32),
				[theme.breakpoints.up('md')]: {
					height: theme.typography.pxToRem(48),
				},
				[theme.breakpoints.up('lg')]: {
					height: theme.typography.pxToRem(72),
				},
			}} />
			Finder
		</Typography>
	)
};

export default Logo;