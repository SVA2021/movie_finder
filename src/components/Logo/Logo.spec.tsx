import { render, screen } from '@testing-library/react';
import Logo from './Logo';

describe('render logo', () => {

	const setUp = () => render(<Logo />);

	it('find movie text', () => {
		setUp();
		expect(screen.getByText(/movie/i)).toBeInTheDocument();
	})
	it('find finder text', () => {
		setUp();
		expect(screen.getByText(/finder/i)).toBeInTheDocument();
	})
	it('find svg icon', () => {
		setUp();
		expect(screen.getByTestId('LocalMoviesIcon')).toBeInTheDocument();
	})
	it('snapshot test', () => {
		setUp();
		expect(screen.getByTestId('logo')).toMatchSnapshot();
	})

	screen.debug();

})