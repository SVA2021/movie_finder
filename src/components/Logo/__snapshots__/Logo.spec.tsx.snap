// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render logo snapshot test 1`] = `
<p
  class="MuiTypography-root MuiTypography-h1 css-352z6l-MuiTypography-root"
  data-testid="logo"
>
  Movie
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1j1iy3m-MuiSvgIcon-root"
    data-testid="LocalMoviesIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M18 3v2h-2V3H8v2H6V3H4v18h2v-2h2v2h8v-2h2v2h2V3h-2zM8 17H6v-2h2v2zm0-4H6v-2h2v2zm0-4H6V7h2v2zm10 8h-2v-2h2v2zm0-4h-2v-2h2v2zm0-4h-2V7h2v2z"
    />
  </svg>
  Finder
</p>
`;
